### PR TITLE
fix(shell): scope .import completion to the typed path prefix

### LINKED ISSUE #597: [PERF] .import completion rescans the entire working tree on every keystroke
## Problem

`.import` completion currently walks the entire working tree on every prompt update and ignores the typed path prefix.

In large repositories this will make completion latency scale with repository size instead of with the subtree the user is targeting.

## Expected behavior

`.import` completion should narrow traversal to the relevant directory or prefix instead of recursively scanning `.` on every keystroke.

## Suggested tests

- Add Go tests that verify prefix-scoped completion behavior.
- Add a benchmark for `.import` completion against a large synthetic directory tree so regressions are measurable.

### PR DESCRIPTION
**Pull request by nao1215**
## Summary

`.import` tab completion walked the entire working tree with `filepath.WalkDir(".")` on every keystroke and ignored the typed prefix, so completion latency scaled with repository size rather than with the subtree the user was targeting. This change scopes traversal to the directory named by the typed prefix.

## Changes

- `shell/shell.go`: rewrite `getFilePathCompletions` to read only the directory named by the typed prefix via `os.ReadDir` instead of recursively walking `.`. Add `splitCompletionPrefix` to split a typed prefix into the directory to scan, the leading text to keep on each suggestion, and the partial name to match. Subdirectories are now offered with a trailing slash (new `msgImportableDir` description) so the path completes one level at a time, the same way a shell completes paths. Hidden entries are skipped unless the user types a leading dot.
- `shell/completion_test.go`: add `TestGetFilePathCompletionsScopedToTypedPrefix` asserting non-recursive, prefix-scoped behavior, add `BenchmarkGetFilePathCompletions` against a large synthetic tree, and update the existing completion tests that encoded the old recursive "show every file" behavior.
- `shell/feature_property_test.go`: add `TestSplitCompletionPrefixProperties`, a property-based check that base+partial reconstructs the input, the partial never spans a separator, and the kept base ends on a separator.
- `shell/commands_test.go`: update the completion edge-case tests to the new scoped behavior.
- `shell/shell_test.go`: widen `newShell` to accept `testing.TB` so the benchmark can build a shell.
- `CHANGELOG.md`: add an Unreleased Bug Fixes entry.

## Design Decisions

Completion now follows the standard shell model: each invocation lists a single directory level and offers subdirectories so the user descends one level at a time. The suggestion `Text` keeps the exact prefix the user typed, which matches how the prompt library applies a completion (`strings.HasPrefix(suggestion.Text, wordBeforeCursor)`), so a full-path suggestion still completes correctly. This intentionally replaces the previous behavior of recursively listing every importable file under the working tree on every keystroke, which was the source of the latency problem and is what the issue asked to remove.

Completion is reached only through the interactive prompt callback; there is no batch or CLI flag path to it (a piped stdin runs in batch mode where the completer never runs), so the behavior is covered by Go tests at the `getCompletions` callback level rather than by a shellspec integration test.

Closes #597

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved file-path completion so suggestions are limited to the directory implied by what you’ve typed, making completions faster and more relevant.
  * Directory suggestions now appear with trailing slashes, and hidden items are only shown when you start typing a hidden name.
  * Completion behavior for nested paths is more accurate, including better handling of partial directory names and platform-specific separators.

### CODE REVIEW

File: shell/shell.go [hunk 1]
```diff
@@ -934,37 +934,66 @@ func (s *Shell) isValidFileForCompletion(filename string) bool {
...
-		// Skip hidden directories and files (unless explicitly requested)
-		// But don't skip the current directory "."
-		if strings.HasPrefix(d.Name(), ".") && d.Name() != "." {
-			if d.IsDir() {
-				return filepath.SkipDir
-			}
-			return nil
+	entries, err := os.ReadDir(readDir)
+	if err != nil {
+		return nil
```
[the remark this answers is not in the document]
Comment (nao1215) [reply shell/shell.go:976]: Good catch. Fixed in dc5a9b7 by normalizing readDir to the local separator (filepath.FromSlash after replacing backslashes) before os.ReadDir, and added a backslash-prefix case to TestGetFilePathCompletionsScopedToTypedPrefix.
